airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ReplicationView.tsx

```diff
@@ -189,11 +181,33 @@ export const ReplicationView: React.FC<ReplicationViewProps> = ({ onAfterSaveSch
     }
   };
 
+  const onRefreshSourceSchema = async () => {
+    if (connectionFormDirtyRef.current) {
+      // The form is dirty so we show a warning before proceeding.
+      openConfirmationModal({
+        title: "connection.updateSchema.formChanged.title",
+        text: "connection.updateSchema.formChanged.text",
+        submitButtonText: "connection.updateSchema.formChanged.confirm",
+        onSubmit: () => {
+          closeConfirmationModal();
+          refreshSourceSchema();
+        },
+      });
+    } else {
+      // The form is not dirty so we can directly refresh the source schema.
+      refreshSourceSchema();
+    }
+  };
+
   const onCancelConnectionFormEdit = () => {
     setSaved(false);
     setActiveUpdatingSchemaMode(false);
   };
 
+  const onDirtyChanges = useCallback((dirty: boolean) => {
+    connectionFormDirtyRef.current = dirty;
```

Comment:
:information_source: We track the dirty state in this component in a ref not state, since we don't require any rerendering of this component if this changes.